smoke/IotEdgeQuickstart/details/Details.cs

```diff
@@ -334,6 +339,45 @@ protected async Task VerifyDataOnIoTHub(string moduleId)
             await eventHubClient.CloseAsync();
         }
 
+        protected async Task VerifyTwinAsync()
+        {
+            if (this.TwinTestFileName.HasValue)
+            {
+                string twinTestJson = JObject.Parse(File.ReadAllText(this.TwinTestFileName.GetOrElse(string.Empty))).ToString();
+
+                var twinTest = JsonConvert.DeserializeObject<TwinTestConfiguration>(twinTestJson);
+
+                Twin currentTwin = await this.context.RegistryManager.GetTwinAsync(this.context.Device.Id, twinTest.ModuleId);
+
+                if (twinTest.Properties?.Desired != null && twinTest.Properties.Desired.Count > 0)
+                {
+                    string patch = JsonConvert.SerializeObject(twinTest, Formatting.Indented);
+                    // Build Patch Object.
+                    await this.context.RegistryManager.UpdateTwinAsync(this.context.Device.Id, twinTest.ModuleId, patch, currentTwin.ETag);
+                }
+
+                if (twinTest.Properties?.Reported != null && twinTest.Properties.Reported.Count > 0)
+                {
+                    TimeSpan retryInterval = TimeSpan.FromSeconds(10);
+                    bool IsValid(TwinCollection currentTwinReportedProperty) => twinTest.Properties.Reported.Cast<KeyValuePair<string, object>>().All(p => currentTwinReportedProperty.Cast<KeyValuePair<string, object>>().Contains(p));
+
+                    using (var cts = new CancellationTokenSource(TimeSpan.FromSeconds(300)))
+                    {
+                        async Task<TwinCollection> Func()
+                        {
+                            // Removing reSharper warning for CTS, Code Block will never exit before the delegate code completes because of using.
+                            // ReSharper disable AccessToDisposedClosure
```

Comment:
This seems a bit risky.. Are you sure it is okay? :)

Reply:
Yeah, I searched and it's a common mistake by resharper. It way that, since we are inside an using block it would be safe, but resharper doesn't recognize that. 
This is where I got it: https://stackoverflow.com/questions/17620430/access-to-disposed-closure-in-c